examples/auto/rust

```diff
@@ -0,0 +1,313 @@
+    ;;
+
+esac
+
+case "$NGX_PLATFORM" in
+
+    OpenBSD:*)
+        # ld: error: undefined symbol: _Unwind_...
+        RUST_LIBS="$RUST_LIBS -lutil"
+        RUST_LIBS="$RUST_LIBS -lexecinfo"
+        RUST_LIBS="$RUST_LIBS -lc++abi"
+    ;;
+
+    win32)
+        case "$NGX_CC_NAME" in
+
+            msvc)
+                # as suggested by rustc --print native-static-libs,
+                # excluding entries already present in CORE_LIBS
+                RUST_LIBS="$RUST_LIBS bcrypt.lib"   # ???
+                RUST_LIBS="$RUST_LIBS ntdll.lib"    # std::io, std::sys::pal::windows
+                RUST_LIBS="$RUST_LIBS userenv.lib"  # std::env::home_dir
+                RUST_LIBS="$RUST_LIBS dbghelp.lib"  # backtrace symbolization
+
+                RUST_TARGET=$RUST_TARGET_ARCH-pc-windows-msvc
+            ;;
+
+            gcc | clang)
+                RUST_LIBS="$RUST_LIBS -lbcrypt"
+                RUST_LIBS="$RUST_LIBS -lntdll"
+                RUST_LIBS="$RUST_LIBS -luserenv"
+                RUST_LIBS="$RUST_LIBS -ldbghelp"
+                # gnullvm on arm64?
+                RUST_TARGET=$RUST_TARGET_ARCH-pc-windows-gnu
+            ;;
+
+        esac
+    ;;
+
+esac
+
+
+# Prepare cargo configuration file
+
+if [ "$NGX_DEBUG" = YES ]; then
+    ngx_cargo_default_profile=ngx-debug
+else
+    ngx_cargo_default_profile=ngx-release
+fi
+
+ngx_cargo_config=$NGX_OBJS/.cargo/config.toml
+ngx_cargo_profile=${ngx_cargo_profile:-$ngx_cargo_default_profile}
+
+mkdir -p "$NGX_OBJS/.cargo"
+
+cat << END                                               > "$ngx_cargo_config"
+
+[profile.ngx-debug]
+inherits = "dev"
+
+[profile.ngx-release]
+inherits = "release"
+lto = "thin"
+strip = "none"
+
+# compatibility with LIBC=-MT set in auto/cc/msvc
+[target.aarch64-pc-windows-msvc]
+rustflags = ["-C", "target-feature=+crt-static"]
+
+[target.i686-pc-windows-msvc]
+rustflags = ["-C", "target-feature=+crt-static"]
+
+[target.x86_64-pc-windows-msvc]
+rustflags = ["-C", "target-feature=+crt-static"]
+
+[env]
+NGINX_BUILD_DIR = { value = ".", force = true, relative = true }
+END
+
+if [ "$NGX_PLATFORM" = win32 ] && command -v cygpath >/dev/null; then
+    printf >> "$ngx_cargo_config" 'NGINX_SOURCE_DIR = "%s"\n' \
+        "$(cygpath -m "$PWD")"
+else
+    printf >> "$ngx_cargo_config" 'NGINX_SOURCE_DIR = "%s"\n' "$PWD"
+fi
+
+
+# Reconstructs path to a static lib built with cargo rustc,
+# relative to the --target-dir
+
+ngx_rust_target_path () {
+    ngx_rust_obj=$(echo "$ngx_rust_target_name" | tr - _)
+
+    case "$NGX_CC_NAME" in
+
+        msvc)
+            ngx_rust_obj=${ngx_rust_obj}.lib
+        ;;
+
+        *)
+            ngx_rust_obj=lib${ngx_rust_obj}.a
+        ;;
+
+    esac
+
+    if [ "$ngx_rust_target_type" = EXAMPLE ]; then
+        ngx_rust_obj=examples/$ngx_rust_obj
+    fi
+
+    echo "${RUST_TARGET:+$RUST_TARGET/}$ngx_cargo_profile/$ngx_rust_obj"
+}
+
+
+# Registers a module in the buildsystem.
+# Expects two variables to be set:
```

Comment:
"Expects two variables to be set:"
->
"In addition to auto/module variables expects two variables to be set:"?